### PR TITLE
Add ability to disable notifications

### PR DESCRIPTION
**Pull request by unixfox**
This PR adds the ability to disable notifications.

-----

In error:
```
Showing last frame. Use --error-trace for full trace.

In src/invidious/channels/channels.cr:229:8

 229 | if preferences.notifications && was_insert
          ^----------
Error: undefined local variable or method 'preferences' for top-level
```

### CODE REVIEW

File: src/invidious/channels/channels.cr [hunk 1]
```diff
@@ -264,7 +264,9 @@ def fetch_channel(ucid, pull_all_videos : Bool)
         # so since they don't provide a published date here we can safely ignore them.
         if Time.utc - video.published > 1.minute
           was_insert = Invidious::Database::ChannelVideos.insert(video)
-          Invidious::Database::Users.add_notification(video) if was_insert
+          if preferences.notifications && was_insert
+            Invidious::Database::Users.add_notification(video)
```
[the remark this answers is not in the document]
Comment (unixfox) [reply src/invidious/channels/channels.cr:268]: We can't do that until we have https://github.com/iv-org/invidious/pull/2878 merged then?

[the remark this answers is not in the document]
Comment (unixfox) [reply src/invidious/channels/channels.cr:268]: Mmh the query you are talking about, is it this one: https://github.com/iv-org/invidious/blob/master/src/invidious/database/users.cr#L100?

We could just temporarily convert the preferences column to a json object then check with a where clause which user has notifications enabled.

**Comment by SamantazFox** [comment]
@unixfox do you mind if I close this PR?

**Comment by unixfox** [comment]
> @unixfox do you mind if I close this PR?

I don't know why you really want to close this PR but if you think you have to do it then you can close it.

**Comment by SamantazFox** [comment]
Related: #1998 #1999

**Comment by unixfox** [comment]
bump

**Comment by SamantazFox** [comment]
I guess this can be closed, now that #3473 has been merged?

**Comment by unixfox** [comment]
> I guess this can be closed, now that #3473 has been merged?

No it's per user here, no globally.

**Pull request closed without merging**